[inductor] Enable mypy checking for torch/_inductor/codegen/triton.py

masnesral opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #109643
* __->__ #109146

Summary: enably mypy chcking for torch/_inductor/codegen/triton.py and make the minimum number of fixes / ignores to get the linter to pass

Test Plan: `lintrunner -a`

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

masnesral committed:
[inductor] Enable mypy checking for torch/_inductor/codegen/triton.py

Summary: enably mypy chcking for torch/_inductor/codegen/triton.py and make the minimum number of fixes / ignores to get the linter to pass

Test Plan: `lintrunner -a`

[ghstack-poisoned]

masnesral committed:
Update on "[inductor] Enable mypy checking for torch/_inductor/codegen/triton.py"


Summary: enably mypy chcking for torch/_inductor/codegen/triton.py and make the minimum number of fixes / ignores to get the linter to pass

Test Plan: `lintrunner -a`

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

masnesral committed:
Update on "[inductor] Enable mypy checking for torch/_inductor/codegen/triton.py"


Summary: enably mypy chcking for torch/_inductor/codegen/triton.py and make the minimum number of fixes / ignores to get the linter to pass

Test Plan: `lintrunner -a`

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

masnesral commented:
@eellison , I'd started on this one last week so I decided to try to finish it.

masnesral commented on torch/_inductor/codegen/triton.py:
```diff
@@ -2279,7 +2285,8 @@ def end_current_reduction_loop():
                         node not in done
                         and fits_in_main_body(other_node)
                         and not (
-                            current_loop_writes & other_node.recursive_predecessors
```
Soooo, `current_write_loops` doesn't exist and this code path must already be broken. I glanced at the history and made a guess that the name should have changed to `current_loop_reduced_writes`, but I don't know what the code is doing exactly so I may be wrong. I see that @Chillee made some changes to this file recently. @Chillee, maybe you can vet this?

masnesral replied:
Or maybe @ezyang? I see your name around these parts.

masnesral replied:
Sorry, scratch that. I guess I see @jansel's name most around this section.

eellison replied:
I think it's defined above, in `generate_node_schedule(self, nodes, numel, rnumel`  Maybe you need `nonlocal current_write_loops`

masnesral replied:
@eellison, sorry my comment has the wrong variable name. My _change_ actually replaces `current_loop_writes` with `current_loop_reduced_writes`. I don't find `current_loop_writes` anywhere in this file.

lezcano replied:
I think this one is on @peterbell10?

peterbell10 replied:
The PR is reverted  already anyway, so a rebase should fix it,

masnesral committed:
Update on "[inductor] Enable mypy checking for torch/_inductor/codegen/triton.py"


Summary: enably mypy chcking for torch/_inductor/codegen/triton.py and make the minimum number of fixes / ignores to get the linter to pass

Test Plan: `lintrunner -a`

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

ezyang commented:
@pytorchbot merge

masnesral committed:
Update on "[inductor] Enable mypy checking for torch/_inductor/codegen/triton.py"


Summary: enably mypy chcking for torch/_inductor/codegen/triton.py and make the minimum number of fixes / ignores to get the linter to pass

Test Plan: `lintrunner -a`

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

masnesral committed:
rebase on "[inductor] Enable mypy checking for torch/_inductor/codegen/triton.py"


Summary: enably mypy chcking for torch/_inductor/codegen/triton.py and make the minimum number of fixes / ignores to get the linter to pass

Test Plan: `lintrunner -a`

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

masnesral commented:
@pytorchbot merge

masnesral committed:
rebase on "[inductor] Enable mypy checking for torch/_inductor/codegen/triton.py"


Summary: enably mypy chcking for torch/_inductor/codegen/triton.py and make the minimum number of fixes / ignores to get the linter to pass

Test Plan: `lintrunner -a`

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

masnesral commented:
@pytorchbot merge

masnesral committed:
rebase onto viable/strict on "[inductor] Enable mypy checking for torch/_inductor/codegen/triton.py"


Summary: enably mypy chcking for torch/_inductor/codegen/triton.py and make the minimum number of fixes / ignores to get the linter to pass

Test Plan: `lintrunner -a`

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

masnesral commented:
@pytorchbot merge

masnesral commented:
@pytorchbot merge
